### PR TITLE
feat: Add LITESTAR_ prefix before WEB_CONCURRENCY env option

### PR DESCRIPTION
**Pull request by kedod**
## Description

- Add `LITESTAR_` prefix before `WEB_CONCURRENCY` env option

### CODE REVIEW

**Comment by cofin** [comment]
I don't think we should add this

File: docs/usage/cli.rst [hunk 1]
```diff
@@ -116,7 +116,10 @@ Options
...
 +-------------------------------------------+----------------------------------------------+----------------------------------------------------------------------------------+
-| ``--wc``\ , ``--web-concurrency``         | ``WEB_CONCURRENCY``                          | The number of concurrent web workers to start [default: 1]                       |
+| ``--wc``\ , ``--web-concurrency``         | ``LITESTAR_WEB_CONCURRENCY``                 | .. deprecated:: 2.8                                                              |
+|                                           |                                              |    Use ``LITESTAR_WEB_CONCURRENCY`` instead ``WEB_CONCURRENCY``                  |
+|                                           |                                              |                                                                                  |
+|                                           |                                              | The number of concurrent web workers to start [default: 1]                       |
```
[the remark this answers is not in the document]
Comment (cofin) [reply docs/usage/cli.rst:122]: I don't think we want to deprecate this as it diverges from the Uvicorn configuration.

[the remark this answers is not in the document]
Comment (JacobCoffee) [reply docs/usage/cli.rst:122]: If we can have both, we can change the `.. deprecated::` directive to `.. versionchanged::` instead

**Comment by provinzkraut** [comment]
> I don't think we should add this. The [Uvicorn](https://www.uvicorn.org/settings/#production) documentation uses `WEB_CONCURRENCY`

We diverge for the other uvicorn settings as well though. I see your point, but I also think we should be consistent with whether we follow uvicorn's naming here :eyes:

**Comment by cofin** [comment]
I think it's also more that the `WEB_CONCURRENCY` variable is a convention that Uvicorn adopted.  

Ultimately, I'll go along with whatever we all decide is best.  It just seems a bit odd that I will have to set both `LITESTAR_WEB_CONCURRENCY` and `WEB_CONCURRENCY` to achieve the same goal I am currently getting with the single `WEB_CONCURRENCY`.

**Comment by peterschutt** [comment]
> It just seems a bit odd that I will have to set both `LITESTAR_WEB_CONCURRENCY` and `WEB_CONCURRENCY` to achieve the same goal I am currently getting with the single `WEB_CONCURRENCY`

The change in this PR looks to me like either `LITESTAR_WEB_CONCURRENCY` or `WEB_CONCURRENCY` would be parsed, correct? So can we just do away with the deprecation and let it be set by either with the `LITESTAR_...` one taking precedence?

**Comment by provinzkraut** [comment]
> > It just seems a bit odd that I will have to set both `LITESTAR_WEB_CONCURRENCY` and `WEB_CONCURRENCY` to achieve the same goal I am currently getting with the single `WEB_CONCURRENCY`
> 
> The change in this PR looks to me like either `LITESTAR_WEB_CONCURRENCY` or `WEB_CONCURRENCY` would be parsed, correct? So can we just do away with the deprecation and let it be set by either with the `LITESTAR_...` one taking precedence?

Wondering if we should do that for all the env vars. That would take care of the uvicorn/general convention naming issue, but also provide an escape hatch for crowded env namespaces.

@cofin wdyt?

**Comment by cofin** [comment]
> > > It just seems a bit odd that I will have to set both `LITESTAR_WEB_CONCURRENCY` and `WEB_CONCURRENCY` to achieve the same goal I am currently getting with the single `WEB_CONCURRENCY`
> > 
> > 
> > The change in this PR looks to me like either `LITESTAR_WEB_CONCURRENCY` or `WEB_CONCURRENCY` would be parsed, correct? So can we just do away with the deprecation and let it be set by either with the `LITESTAR_...` one taking precedence?
> 
> Wondering if we should do that for all the env vars. That would take care of the uvicorn/general convention naming issue, but also provide an escape hatch for crowded env namespaces.
> 
> @cofin wdyt?

Sorry for the delay in reply on this one.  I think this is a good approach.

**Review by cofin — APPROVED** [review]

**Pull request merged**